test/OpenTelemetry.Instrumentation.Http.Tests/HttpTestData.cs

```diff
@@ -73,7 +73,60 @@ public class HttpOutTestCase
 
             public bool? SpanStatusHasDescription { get; set; }
 
+            [JsonConverter(typeof(DictionaryStringStringConverter))]
             public Dictionary<string, string> SpanAttributes { get; set; }
         }
+
+        /// <summary>
+        /// Convert string-number to string-string pair.
+        /// </summary>
+        private class DictionaryStringStringConverter : JsonConverter<Dictionary<string, string>>
```

Comment:
There is some inconsistency with the way test cases are written in this project. I think we can rid of this class if we just consistently pass in the values for `"http.status_code"` in `string` format for all the test cases. For example: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/OpenTelemetry.Instrumentation.Http.Tests/http-out-test-cases.json#L200

Reply:
Yeah. If they are all string, we won't need this class. But I'm not sure whether we have a way to enforce consistency. Using this class could make the Span Attributes less sensitive and more extensible I think.

Reply:
More code = More maintenance

Firstly, these are test files. This is not going to be used by any user so there is no significant benefit to allowing someone to casually add statusCode in a number format. Secondly, it's enforced by test failure. If you remove this newly added class the tests won't pass until you fix the statusCode format.

Reply:
> More code = More maintenance
> 
> Firstly, these are test files. This is not going to be used by any user so there is no significant benefit to allowing someone to casually add statusCode in a number format. Secondly, it's enforced by test failure. If you remove this newly added class the tests won't pass until you fix the statusCode format.

Ok. You convinced me. Removed. Please review again.

Reply:
Looks good. Please fix the failing test.

Reply:
Fixed.